### PR TITLE
[@types/atom] Add a missing field to the options

### PR DESCRIPTION
**Pull request by illright**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/atom/autocomplete-plus/wiki/Provider-API#defining-a-provider

This will also reorder the options to match the order in the docs for easier comparison.

### CODE REVIEW

**Review by aminya — COMMENTED** [review]
Could you keep the order of the methods? It is hard to follow the changes

**Comment by illright** [comment]
@aminya I could, but it makes maintenance more difficult in the long run, IMHO. For now I can assure you that the only thing I actually changed in the code is [this](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50407/files#diff-777aac71c6a6427573daa4d10dd044bb50b4d7b03f8376a314461d304ae996d6R148-R150), everything else is just reordering. If you want me to, I can revert the order to the original one.

**Review by aminya — APPROVED** [review]

**Review by lierdakil — APPROVED** [review]
LGTM

**Comment by andrewbranch** [comment]
CI is confused 🤔

**Comment by andrewbranch** [comment]
Azure DevOps looks good... hopefully GitHub Actions just hasn’t had its morning coffee yet
